library_builders.sh

```diff
@@ -9,13 +9,27 @@ CPU_COUNT="${CPU_COUNT:-2}"
 function install_buildessentials {
     if [ -e buildessentials-stamp ]; then return; fi
 
-    # Cleanup:
-    #   - Travis-CI macOS ships a pre-installed HDF5
     if [ "$(uname -s)" = "Darwin" ]
     then
+        # Cleanup:
+        #   - Travis-CI macOS ships a pre-installed HDF5
         brew unlink hdf5 || true
         brew uninstall --ignore-dependencies hdf5 || true
         rm -rf /usr/local/Cellar/hdf5
+
+        # we need qemu to configure cross-compiles for HDF5
+        if [[ "${CMAKE_OSX_ARCHITECTURES-}" == *"arm64"* ]]; then
+            brew install qemu libvirt || true
+
+            # TODO: patch libvirt?
+            #https://raw.githubusercontent.com/yoonsikp/vm_configs/master/libvirt.rb
+
+            # https://www.naut.ca/blog/2021/12/09/arm64-vm-on-macos-with-libvirt-qemu/
+            echo 'security_driver = "none"' >> /opt/homebrew/etc/libvirt/qemu.conf
+            echo "dynamic_ownership = 0" >> /opt/homebrew/etc/libvirt/qemu.conf
+            echo "remember_owner = 0" >> /opt/homebrew/etc/libvirt/qemu.conf
+            brew services start libvirt || true
+        fi
```

Comment:
`qemu-system-aarch64` does not work, we would need `qemu-aarch64`, which I cannot find anywhere for mac.
```suggestion
```